src/app/components/login-card/login-card.component.ts

```diff
@@ -1,15 +1,87 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Router } from '@angular/router';
+import { loginUser } from 'src/app/models/loginUser';
+import { User } from 'src/app/models/user';
+import { AuthenticationService } from 'src/app/services/auth.service';
 
 @Component({
   selector: 'app-login-card',
   templateUrl: './login-card.component.html',
-  styleUrls: ['./login-card.component.scss']
+  styleUrls: ['./login-card.component.scss'],
 })
 export class LoginCardComponent implements OnInit {
+  loginForm!: FormGroup;
+  isInputValid: boolean = false;
+  loginStatus: boolean = false;
+  loginMessage: string = '';
+  error = '';
 
-  constructor() { }
+  responseUserObject?: loginUser;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private loginService: AuthenticationService,
+    private snackBar: MatSnackBar,
+    private router: Router
+  ) {if (this.loginService.currentUserValue) { 
```

Comment:
Pay more attention to the functions format

Reply:
Not entirely sure how to solve this

Reply:
) {
      if (this.loginService.currentUserValue) {
          this.router.navigate(['/events-list']);
       }
 }

Reply:
Oh, ok I got it now